[Improvement][Server] When the Worker turns down, the MasterServer cannot handle the Remove event correctly and throws NPE

**Describe the question**
When the Worker turns down, the MasterServer cannot handle the Remove event correctly and throws an NPE exception:
```java
[ERROR] 2021-07-10 00:27:52.736 org.apache.curator.framework.recipes.cache.TreeCache:[779] - 
java.lang.NullPointerException: null
	at org.apache.dolphinscheduler.server.master.registry.MasterRegistryDataListener.handleWorkerEvent(MasterRegistryDataListener.java:83)
	at org.apache.dolphinscheduler.server.master.registry.MasterRegistryDataListener.notify(MasterRegistryDataListener.java:47)
	at org.apache.dolphinscheduler.spi.register.ListenerManager.dataChange(ListenerManager.java:63)
	at org.apache.dolphinscheduler.plugin.registry.zookeeper.ZookeeperRegistry.lambda$subscribe$0(ZookeeperRegistry.java:166)
	at org.apache.curator.framework.recipes.cache.TreeCache$2.apply(TreeCache.java:760)
	at org.apache.curator.framework.recipes.cache.TreeCache$2.apply(TreeCache.java:754)
	at org.apache.curator.framework.listen.ListenerContainer$1.run(ListenerContainer.java:100)
	at org.apache.curator.shaded.com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:30)
	at org.apache.curator.framework.listen.ListenerContainer.forEach(ListenerContainer.java:92)
	at org.apache.curator.framework.recipes.cache.TreeCache.callListeners(TreeCache.java:753)
	at org.apache.curator.framework.recipes.cache.TreeCache.access$1900(TreeCache.java:75)
	at org.apache.curator.framework.recipes.cache.TreeCache$4.run(TreeCache.java:865)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:266)
	at java.util.concurrent.FutureTask.run(FutureTask.java)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
	
```

**What are the current deficiencies and the benefits of improvement**
```java
public class MasterRegistryDataListener implements SubscribeListener {

    private static final Logger logger = LoggerFactory.getLogger(MasterRegistryDataListener.class);
    
   @Resource
    private MasterRegistryClient masterRegistryClient;
```

The `MasterRegistryDataListener` class will be used by `MasterRegistryClient`, as shown below:
```java
// line 123
registryClient.subscribe(REGISTRY_DOLPHINSCHEDULER_NODE, new MasterRegistryDataListener());
```
**The @Resource annotation will not take effect in the current case, it would be assigned to null**

**Which version of DolphinScheduler:**
latest dev branch

--- a/dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/registry/MasterRegistryDataListener.java
+++ b/dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/registry/MasterRegistryDataListener.java
@@ -22,10 +22,9 @@ import static org.apache.dolphinscheduler.common.Constants.REGISTRY_DOLPHINSCHED
 
 import org.apache.dolphinscheduler.common.Constants;
 import org.apache.dolphinscheduler.common.enums.NodeType;
+import org.apache.dolphinscheduler.service.bean.SpringApplicationContext;
 import org.apache.dolphinscheduler.spi.register.DataChangeEvent;
 import org.apache.dolphinscheduler.spi.register.SubscribeListener;
-
-import javax.annotation.Resource;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
@@ -34,8 +33,12 @@ public class MasterRegistryDataListener implements SubscribeListener {
 
     private static final Logger logger = LoggerFactory.getLogger(MasterRegistryDataListener.class);
 
-    @Resource
-    MasterRegistryClient masterRegistryClient;
+    private MasterRegistryClient masterRegistryClient;
+
+    public MasterRegistryDataListener() {
+        masterRegistryClient = SpringApplicationContext.getBean(MasterRegistryClient.class);
+    }
+
 
     @Override
     public void notify(String path, DataChangeEvent event) {

